diff_drive_controller/include/diff_drive_controller/odometry.hpp

```diff
@@ -37,10 +37,25 @@ class Odometry
 public:
   explicit Odometry(size_t velocity_rolling_window_size = 10);
 
+  [[deprecated]]
   void init(const rclcpp::Time & time);
+  [[deprecated(
+    "Replaced by bool updateFromPos(const double left_pos, const double right_pos, const "
+    "rclcpp::Time & time).")]]
   bool update(double left_pos, double right_pos, const rclcpp::Time & time);
+  [[deprecated(
+    "Replaced by bool updateFromVel(const double left_vel, const double right_vel, const "
+    "rclcpp::Time & time).")]]
   bool updateFromVelocity(double left_vel, double right_vel, const rclcpp::Time & time);
+  [[deprecated(
+    "Replaced by bool tryUpdateOpenLoop(const double linear_vel, const double angular_vel, const "
+    "rclcpp::Time "
+    "& time).")]]
   void updateOpenLoop(double linear, double angular, const rclcpp::Time & time);
+
+  bool updateFromPos(const double left_pos, const double right_pos, const double dt);
+  bool updateFromVel(const double left_vel, const double right_vel, const double dt);
+  bool tryUpdateOpenLoop(const double linear_vel, const double angular_vel, const double dt);
```

Comment:
- no need for const qualifier for non-ref datatypes. 
- let's change the code style to snake case for new methods.